### PR TITLE
README: update for concept exercises

### PR DESCRIPTION
**Pull request by pwadsworth**
Adds concepts and concept exercises entries. 

Corrects folder structure to include concepts.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -196,4 +230,4 @@ code worse, you can [suppress it](https://github.com/ndmitchell/hlint#customizin
 with annotations in the source file.
 
 ## Automated Test Runner
-We have a [test runner](https://github.com/exercism/haskell-test-runner) to automatically run tests on Haskell solutions submitted to [exercism](exercism.org).
+We have a [test runner](https://github.com/exercism/haskell-test-runner) to automatically run tests on Haskell solutions submitted to [exercism](exercism.org).
```
[the remark this answers is not in the document]
Comment (petertseng) [reply README.md:233]: (reason: https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline)

File: README.md [hunk 2]
```diff
@@ -67,38 +74,65 @@ The [track anatomy documentation](https://github.com/exercism/docs/blob/master/l
...
-  │ ├── package.yaml
-  │ ├── stack.yaml
-  │ ├── src
-  │ │ └── Accumuĺate.hs
-  │ ├── test
-  │ │ └── Tests.hs
-  │ └── .meta
-  │   ├── examples
-  │   │ └── success-standard
+  └── concept
```
Comment (petertseng) [inline README.md:98]: vertical line extending under `exercises` isn't aligned with first character (`e`) of `exercises`.

I gather that the point of realigning this section is to consistently make the vertical line under a directory aligned with its first character, but the rule was broken here. Either revert the change and do no reformatting, or apply the new rule to `exercises` as well.

File: README.md [hunk 3]
```diff
@@ -67,38 +74,65 @@ The [track anatomy documentation](https://github.com/exercism/docs/blob/master/l
...
+  │   │   └── .meta
+  │   │       ├── config.json
+  │   │       ├── design.md
+  │   │       └── exemplar
+  │   │           ├── package.yaml
+  │   │           └── src
+  │   │               └── AnnalynsInfiltration.hs
+  │   └── ...
+  └── practice
+    ├── accumulate
```
Comment (petertseng) [inline README.md:119]: vertical line extending under `practice` isn't aligned with first character (`p`) of `practice`.

I gather that the point of realigning this section is to consistently make the vertical line under a directory aligned with its first character, but the rule was broken here. Either revert the change and do no reformatting, or apply the new rule to `practice` as well.

File: README.md [hunk 4]
```diff
@@ -67,38 +74,65 @@ The [track anatomy documentation](https://github.com/exercism/docs/blob/master/l
...
+    │   ├── package.yaml
+    │   ├── stack.yaml
+    │   ├── src
+    │   │   └── Accumuĺate.hs
+    │   ├── test
+    │   │   └── Tests.hs
+    │   └── .meta
+    │     ├── examples
```
Comment (petertseng) [inline README.md:127]: vertical line extending under `.meta` isn't aligned with first character (`.`) of `.meta`.

I gather that the point of realigning this section is to consistently make the vertical line under a directory aligned with its first character, but the rule was broken here. Either revert the change and do no reformatting, or apply the new rule to `.meta` as well.

**Comment by pwadsworth** [comment]
Finally starting to get the hang of  putting PRs. in order Thanks for the patience!

File: README.md [hunk 5]
```diff
@@ -67,38 +74,65 @@ The [track anatomy documentation](https://github.com/exercism/docs/blob/master/l
...
-  │ ├── package.yaml
-  │ ├── stack.yaml
-  │ ├── src
-  │ │ └── Accumuĺate.hs
-  │ ├── test
-  │ │ └── Tests.hs
-  │ └── .meta
-  │   ├── examples
-  │   │ └── success-standard
+  ├── concept
```
Comment (petertseng) [inline README.md:98]: vertical line extending below `exercises` still is not aligned with the first `e` of `exercises`

**Review by petertseng — COMMENTED** [review]
Of course. One of the things I find in open source is that patience and understanding is needed from both sides, because most of it is volunteer work. Just as I have to guide all contributors to do their best work, I have to also be thankful that they're willing to put in the time at all and put up with the changes I ask for, even if I may seem ungrateful.

File: README.md [hunk 6]
```diff
@@ -9,6 +9,8 @@ Exercism exercises in Haskell
   * [Report or fix a bug](#report-or-fix-a-bug)
   * [Review issues and pull requests](#review-issues-and-pull)
   * [Port or create an exercise](#port-or-create-an-exercise)
+  * [Port or create a concept](#port-or-create-a-concepts)
```
Comment (petertseng) [inline README.md:12]:
```suggestion
  * [Port or create a concept](#port-or-create-a-concept)
```

concepts wouldn't link, because the heading title is singular

**Comment by pwadsworth** [comment]
Corrected link and alignment. Tripled checked and found no other problems 🤞

**Review by petertseng — APPROVED** [review]

**Comment by petertseng** [comment]
Great, thanks

**Comment by pwadsworth** [comment]
@petertseng: The commits you made left my fork 5 commits ahead of exercism/haskell (plus 1 when I merged), but there are no changes in the files. What is the correct way forward? do I make another pull request with the 6 commits and no file changes or reset (read delete and fork again) my fork?

**Comment by pwadsworth** [comment]
I am going to go with option 2. There are other differences with my local repos and I lost track of which one is which.

**Comment by petertseng** [comment]
You could reset, but the reset I had in mind doesn't involve deleting and forking again; resetting involves using `git reset`, potentially `git reset --hard`. When a word already means something in a particular context ("reset" has a specific meaning in the context of Git), using it to mean something other than its usual meaning should be avoided.
